src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -1235,6 +1235,9 @@ private[internal] trait TypeMaps {
             if (decl.hasAllFlags(METHOD | MODULE))
               // HACK: undo flag Uncurry's flag mutation from prior run
               decl.resetFlag(METHOD | STABLE)
+            if (decl.hasAllFlags(METHOD | JAVA_ANNOTATION))
+              // JAVA_ANNOTATION is a marker; hasAnnotation(inheritSignatureClass) causes cycles
+              decl.setFlag(Flags.OVERRIDE).resetFlag(Flags.PRIVATE | Flags.ARTIFACT | Flags.JAVA_ANNOTATION)
           }
```

Comment:
Maybe slightly less of a hack to leave the `InheritedSignature` symbol attachment on the method and use that as the condition here?

Reply:
👍  thanks